sdk/core/Azure.Core/src/Shared/OperationInternalOfT.cs

```diff
@@ -288,6 +288,9 @@ private class FinalOperation : IOperation<T>
         {
             public ValueTask<OperationState<T>> UpdateStateAsync(bool async, CancellationToken cancellationToken)
                 => throw new NotSupportedException("The operation has already completed");
+
+            // Unreachable path. _operation.GetRehydrationToken() is never invoked.
+            public RehydrationToken? GetRehydrationToken() => null;
```

Comment:
Should this throw?

Reply:
It doesn't matter much as we never invoke this method. As a private class, `FinalOperation` is only accessed from `OperationInternal<T>`, which doesn't have a `GetRehydrationToken` method.

Reply:
From a logical perspective, I don't think it should throw. It's reasonable to request a rehydration token from a final operation if you want to consult its status and result again later with a service call.

We don't really need to implement it here though because we any `FinalOperation` instance we create never leaves the `OperationInternal<T>` object.

Reply:
> Sure, but by throwing we can ensure that it is never actually called unintentionally.

That's reasonable. We can throw here and add a different error message.

Reply:
Done.